app/controllers/teachers/schools_controller.rb

```diff
@@ -10,13 +10,25 @@ class SchoolsController < ApplicationController
     def show
       redirect_to enrol_path unless @school.active? || (current_user && current_user.manages_school?(@school.id))
 
-      @charts = [:teachers_landing_page_electricity, :teachers_landing_page_gas]
+      setup_charts
       setup_dashboard_alert
       setup_activity_suggestions
     end
 
   private
 
+    def setup_charts
+      @charts = {}
+
+      if @school.meters_for_supply(:electricity).any?
```

Comment:
Should this be in the config too? Saves us querying meters again.

Reply:
Yep, we can probably remove a lot of these 'meters_for_supply' and other such queries if we persist it - also checking that we have any readings rather than just a meter.